src/Time/Actions/SelfStartActions.hpp

```diff
@@ -0,0 +1,396 @@
+///
+/// Each time the state is reset the slab number is increased by one.
+/// This ensures that the evaluations are considered to be ordered in
+/// their evaluation order, even though they are not monotonic in
+/// time.  When the slab number reaches zero the initialization
+/// procedure is complete and history appropriate for use for an
+/// integrator of order \f$N+1\f$ has been generated.
+///
+/// The self-start procedure performs all its evaluations before the
+/// end of the first time step of the main evolution.  It is important
+/// that none of the early steps fall at the same time as the
+/// self-start history values, so the main evolution should not
+/// decrease its step size on the first step after the procedure.
+/// Additionally, the history times will not be monotonicly increasing
+/// until \f$N\f$ steps have been taken.  The local-time-stepping
+/// calculations require monotonic time, so local time-stepping should
+/// not be initiated until the self-start values have expired from the
+/// history.  These restrictions on step-size changing are checked in
+/// the TimeStepper::can_change_step_size method.
+namespace SelfStart {
+/// Self-start tags
+namespace Tags {
+/// \ingroup TimeGroup
+/// The initial value of a quantity.  The contents are stored in a
+/// tuple to avoid putting duplicate tensors into the DataBox.
+template <typename Tag>
+struct InitialValue : db::PrefixTag, db::SimpleTag {
+  static std::string name() noexcept {
+    return "InitialValue(" + Tag::name() + ")";
+  }
+  using tag = Tag;
+  using type = std::tuple<db::item_type<Tag>>;
+};
+}  // namespace Tags
+
+/// Self-start actions
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \ingroup TimeGroup
+/// Prepares the evolution for time-stepper self-starting.
+///
+/// Stores the initial values of the variables and time step and sets
+/// an appropriate step for self-starting.
+///
+/// Uses:
+/// - ConstGlobalCache: nothing
+/// - DataBox:
+///   - Tags::TimeId
+///   - Tags::TimeStep
+///   - variables_tag
+///
+/// DataBox changes:
+/// - Adds:
+///   - SelfStart::Tags::InitialValue<Tags::TimeStep>
+///   - SelfStart::Tags::InitialValue<variables_tag>
+/// - Removes: nothing
+/// - Modifies: Tags::TimeStep
+struct Initialize {
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTags>& box,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using variables_tag = typename Metavariables::system::variables_tag;
+
+    // The self-start procedure must take place within one slab, and
```

Comment:
This comment was extremely useful. I think so much so that it would be nice to have in the doxygen documentation for the struct.